src/components/Color.js

```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styles from './Color.css';
+
+const colorToHex = color => {
+  return `${color.toString(16).padStart(2, 0)}`.toUpperCase();
+};
+
```

Comment:
You receive the prop ```{ name: 'red', rgb: { red: 255, green: 0, blue: 0 } }``` which you then destructure on line 13...which gives you ```name: 'red';``` and ```rgb: { red: 255, green: 0, blue: 0 }```. You don't need to destructure on line 9 because you are passing the correct data structure (rgb: object) for the function on line 14. I'd be happy to clarify further if you still have questions.